packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,267 @@
+                            ...rules[index],
+                            awsArn: e.target.value,
+                          })
+                        }
+                        placeholder="arn:aws:sts::111111111111:assumed-role/teleport-node-role/i-*"
+                        value={rule.awsArn}
+                      />
+                    </>
+                  )}
+                </RuleBox>
+              ))}
+            </Box>
+            <Box
+              onClick={() =>
+                setRules([
+                  ...rules,
+                  { awsAccount: '', awsArn: '', isCollapsed: false },
+                ])
+              }
+            >
+              <ButtonAddRule>+ Add new rule</ButtonAddRule>
+            </Box>
+            {token && (
+              <>
+                <Text mt="3" mb="2" bold typography="h5">
+                  Script Generated
+                </Text>
+                <Text>
+                  Start teleport with the following parameters. The script will
+                  be valid for{' '}
+                  <Text bold as={'span'}>
+                    {expiry}.
+                  </Text>
+                </Text>
+                <Box>
+                  <TextSelectCopy
+                    mt="2"
+                    text={`teleport start --roles=node --token=${token} --auth-server=${host} --join-method=iam`}
+                  />
+                </Box>
+              </>
+            )}
+          </DialogContent>
+          <DialogFooter>
+            <ButtonPrimary
+              mr="3"
+              disabled={attempt.status === 'processing'}
+              onClick={() => handleGenerate(validator)}
+            >
+              {token ? 'Regenerate Script' : 'Generate Script'}
+            </ButtonPrimary>
+            <ButtonSecondary onClick={onClose}>Close</ButtonSecondary>
+          </DialogFooter>
+        </>
+      )}
+    </Validation>
+  );
+}
+
+const RuleBox = styled(Box)`
+  margin: 4px 0 12px 0;
+  position: relative;
+
+  &:last-of-type {
+    margin-bottom: 8px;
+  }
+`;
+
+const ButtonAddRule = styled(ButtonLink)`
+  font-weight: bold;
+  font-size: 14px;
+  color: white;
+  text-decoration: none;
+`;
+
+const ButtonEditRule = styled(ButtonLink)`
+  position: absolute;
+  top: 12px;
+  right: 12px;
+  font-size: 16px;
+  text-decoration: none;
+`;
+
+const ButtonRemoveRule = styled(ButtonLink)`
+  position: absolute;
+  top: 12px;
+  right: 12px;
+`;
+
+export const AWS_ACC_ID_REGEXP = /^\d{12}$/;
+const requiredAwsAccount = (value, rule: Rule) => () => {
+  if (!rule.awsAccount && !rule.awsArn) {
+    return {
+      valid: false,
+      message: 'Rule cannot be empty',
+    };
+  }
+
+  // no need for an AWS account if an ARN is set
+  if (!value) {
+    return { valid: true };
+  }
+
+  const isValidId = value.match(AWS_ACC_ID_REGEXP);
+  if (!isValidId) {
+    return {
+      valid: false,
+      message: 'AWS account must have 12 digits',
```

Comment:
Just going from the regex this error is a bit better...
```suggestion
      message: 'AWS account must be 12 digits',
```

Reply:
Thanks!